service/service.go

```diff
@@ -1,312 +1,276 @@
 	}()
 
 	signalChan := make(chan os.Signal, 1)
 	signal.Notify(signalChan, syscall.SIGINT, syscall.SIGTERM)
 	<-signalChan
 
-	log.Println("shotdown signal received, exiting")
+	logger.Info("shotdown signal received, exiting")
 	grpcServer.GracefulStop()
 }
+
+func validatePostgresConfig(logger *zap.Logger, c *repository.PostgresConfig) {
+	if c.Schema == "" {
+		logger.Info("error during reading env. variable", zap.String("default value is used ", defaultDbScheme))
+		c.Schema = defaultDbScheme
+	}
+	if c.DSN == "" {
+		logger.Info("error during reading env. variable", zap.String("default value is used ", defaultDbDSN))
+		c.DSN = defaultDbDSN
+	}
+	//if c.dbHost == "" {
+	//	logger.Info("error during reading env. variable", zap.String("default value is used ", defaultDbHost))
+	//	c.dbHost = defaultDbHost
+	//}
+	//if c.dbPort == "" {
+	//	logger.Info("error during reading env. variable", zap.String("default value is used ", defaultDbPort))
+	//	c.dbPort = defaultDbPort
+	//}
+	//if c.dbUser == "" {
+	//	logger.Info("error during reading env. variable", zap.String("default value is used ", defaultDbUser))
+	//	c.dbUser = defaultDbUser
+	//}
+	//if c.dbPassword == "" {
+	//	logger.Info("error during reading env. variable", zap.String("default value is used ", defaultDbPassword))
+	//	c.dbPassword = defaultDbPassword
+	//}
+	//if c.dbName == "" {
+	//	logger.Info("error during reading env. variable", zap.String("default value is used ", defaultDbName))
+	//	c.dbName = defaultDbName
+	//}
+	if c.MaxOpenedConnectionsToDb == 0 {
+		logger.Info("maxOpenedConnectionsToDb = 0", zap.Int("default value is used ", defaultMaxOpenedConnectionsToDb))
+		c.MaxOpenedConnectionsToDb = defaultMaxOpenedConnectionsToDb
+	}
+	if c.MaxIdleConnectionsToDb == 0 {
+		logger.Info("maxIdleConnectionsToDb = 0", zap.Int("default value is used ", defaultMaxIdleConnectionsToDb))
+		c.MaxIdleConnectionsToDb = defaultMaxIdleConnectionsToDb
+	}
+	if c.MbConnMaxLifetimeMinutes == 0 {
+		logger.Info("mbConnMaxLifetimeMinutes = 0", zap.Int("default value is used ", defaultmbConnMaxLifetimeMinutes))
+		c.MbConnMaxLifetimeMinutes = defaultmbConnMaxLifetimeMinutes
+	}
+}
+
+func initPostgresConfig(logger *zap.Logger) *repository.PostgresConfig {
+	c := new(repository.PostgresConfig)
+	c.Schema = os.Getenv(pdbScheme)
+	c.DSN = os.Getenv(pdbDSN)
+	//c.dbHost = os.Getenv(pdbHost)
+	//c.dbPort = os.Getenv(pdbPort)
+	//c.dbUser = os.Getenv(pdbUser)
+	//c.dbPassword = os.Getenv(pdbPassword)
+	//c.dbName = os.Getenv(pdbName)
+	var err error
+	c.MaxOpenedConnectionsToDb, err = strconv.Atoi(os.Getenv(maxOpCon))
+	if err != nil {
+		logger.Info("error during reading env. variable. Could not convert from string to int", zap.Error(err))
+	}
+	c.MaxIdleConnectionsToDb, err = strconv.Atoi(os.Getenv(maxIdleCon))
+	if err != nil {
+		logger.Info("error during reading env. variable. Could not convert from string to int", zap.Error(err))
```

Comment:
Seems like this messages are the same for different functions? maybe, they can be passed like constants?